src/runtime/disco/loader.cc

```diff
@@ -93,9 +93,18 @@ inline std::vector<ShapeTuple::index_type> ShardShape(const ShapeTuple& shape, i
   return result;
 }
 
+class ShardInfoCompare {
+  public:
+    bool operator()(const ShardLoaderObj::ShardInfo& a, const ShardLoaderObj::ShardInfo& b) const {
+      int a_id = std::stoi(a.param->name.substr(6));
+      int b_id = std::stoi(b.param->name.substr(6));
+      return a_id < b_id;
```

Comment:
you don't need an actual "sort", instead you may retrieve the parameter names in ndarray-cache-metadata and calculate their names correspondingly by "params_0", "params_1", etc